### PR TITLE
chore: replace webpack to rollup for UMD bundle in rest-api-client

### PR DESCRIPTION
**Pull request by shisama**
## Why

Bundle size. The minified bundle file with Rollup are smaller than the ones with webpack.

I tried Rollup and webpack to compare the sizes.
The results are below.

-|KintoneRestAPIClient.js|KintoneRestAPIClient.min.js
---|---|---
webpack | 749698 | 100007
Rollup | 844037 |  85570

In addition, I think Rollup is simpler.

## What

Replacing webpack to Rollup for UMD bundle.

## How to test

Run `yarn build` and try importing a bundled file into your kintone environment.

## Checklist

- [x] Read [CONTRIBUTING.md](https://github.com/kintone/js-sdk/blob/master/CONTRIBUTING.md)
- [ ] Updated documentation if it is required.
- [ ] Added tests if it is required.
- [x] Passed `yarn lint` and `yarn test` on the root directory.

### CODE REVIEW

File: packages/rest-api-client/rollup.config.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import babel from "@rollup/plugin-babel";
+
+const extensions = [".ts", ".js"];
+
+export default (argv) => {
+  const isProd = argv.mode === "production";
+  const plugins = [
+    // typescript({
+    //   declaration: false,
+    // }),
```
Comment (shisama) [inline packages/rest-api-client/rollup.config.js:19]: I first tried `plugin-typescript` but an error occurs due to the lack of Node.js Core API shim. I tried 'plugin-babel' and succeeded in running properly. I don't know why I could. Please tell me or point out if you know about something about this problem.

File: packages/rest-api-client/tsconfig.json [hunk 2]
```diff
@@ -12,7 +12,7 @@
     // "declarationMap": true,                /* Generates a sourcemap for each corresponding '.d.ts' file. */
     // "sourceMap": true,                     /* Generates corresponding '.map' file. */
     // "outFile": "./",                       /* Concatenate and emit output to single file. */
-    "outDir": "./lib",                        /* Redirect output structure to the directory. */
+    // "outDir": "./lib",                        /* Redirect output structure to the directory. */
```
Comment (shisama) [inline packages/rest-api-client/tsconfig.json:15]: UMD build doesn't need transpiled files in 'outDir'. If this is designated, transpiled files are created when Rollup is executed. So, I commented out this option and designate outDir in package.json.

File: packages/rest-api-client/package.json [hunk 3]
```diff
@@ -58,9 +58,19 @@
...
     "@types/js-base64": "^2.3.2",
     "@types/qs": "^6.9.4",
+    "rollup": "^2.23.0",
+    "rollup-plugin-node-builtins": "^2.1.2",
+    "rollup-plugin-node-globals": "^1.4.0",
+    "rollup-plugin-replace": "^2.2.0",
+    "rollup-plugin-terser": "^6.1.0",
     "ts-loader": "^8.0.2",
     "webpack": "^4.44.1",
     "webpack-cli": "^3.3.12"
```
Comment (shisama) [inline packages/rest-api-client/package.json:76]: This is still experimental. If this PR is approved, I will remove webpack dependencies.

File: packages/rest-api-client/src/index.browser.ts [hunk 4]
```diff
@@ -2,7 +2,11 @@ import "core-js/features/promise";
...
 import { injectPlatformDeps } from "./platform/";
 import * as browserDeps from "./platform/browser";
+import { KintoneRestAPIClient } from "./KintoneRestAPIClient";
 
 injectPlatformDeps(browserDeps);
 
 export { KintoneRestAPIClient } from "./KintoneRestAPIClient";
+
+// @ts-ignore
+window.KintoneRestAPIClient = KintoneRestAPIClient;
```
Comment (koba04) [inline packages/rest-api-client/src/index.browser.ts:12]: @shisama 
This file is also used as an entry point for bundling with webpack, so we should avoid leaking global variables.
I guess that we can add the global variable through a rollup configuration like this.

```diff
diff --git a/packages/rest-api-client/rollup.config.js b/packages/rest-api-client/rollup.config.js
index 4001eb8d..be4409c0 100644
--- a/packages/rest-api-client/rollup.config.js
+++ b/packages/rest-api-client/rollup.config.js
@@ -40,9 +40,10 @@ export default (argv) => {
   return {
     input: "./src/index.browser.ts",
     output: {
+      extend: true,
       file: `./umd/KintoneRestAPIClient${isProd ? ".min" : ""}.js`,
       format: "umd",
-      name: "KintoneRestAPIClient",
+      name: "window",
       sourcemap: isProd ? false : "inline",
     },
     plugins,
```

Comment (shisama) [reply packages/rest-api-client/src/index.browser.ts]: @koba04 Thank you for your help. I didn't know about [output.extend](https://rollupjs.org/guide/en/#outputextend). I fixed this.

File: packages/rest-api-client/rollup.config.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  const isProd = argv.mode === "production";
+  const plugins = [
+    babel({
+      babelHelpers: "bundled",
+      presets: [
+        [
+          "@babel/preset-env",
+          {
+            corejs: 3,
+            targets: { browsers: ["> 0.25%", "IE 11", "not op_mini all"] },
```
Comment (shisama) [inline packages/rest-api-client/rollup.config.js:23]: point out kintone's suported browsers explicitly. https://get.kintone.help/general/en/user/list_start/webbrowser.html

**Comment by shisama** [comment]
@koba04 
Thanks for your review. I fixed them you had pointed out. Please look at them again.

**Comment by koba04** [comment]
@shisama 👍  You can also remove `ts-loader` from dependencies.

**Comment by shisama** [comment]
@koba04 Thanks for your good catching! I removed it.

**Pull request merged**